### PR TITLE
Adds cert server to buf from buf/core

### PR DESCRIPTION
**Pull request by smaye81**
This adds the `private/pkg/cert/certserver` package to this repo.  This package is a wholesale copy from the `certserver` located in `buf/core`.  The `certserver` in `buf/core` will be removed in a followup PR.

This is needed for configuration when building Connect clients.

### CODE REVIEW

File: private/pkg/cert/certserver/certserver.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+type ExternalServerTLSConfig struct {
+	Use                string `json:"use,omitempty" yaml:"use,omitempty"`
+	ServerCertFilePath string `json:"server_cert_file_path,omitempty" yaml:"server_cert_file_path,omitempty"`
+	ServerKeyFilePath  string `json:"server_key_file_path,omitempty" yaml:"server_key_file_path,omitempty"`
+}
+
+// NewServerTLSConfig creates a new *tls.Config from the ExternalTLSConfig
+//
+// The default is to use a local TLS config at ${configDirPath}/tls/{cert,key}.pem.
+func NewServerTLSConfig(
```
Comment (bufdev) [inline private/pkg/cert/certserver/certserver.go:36]: Just making sure - why is this needed for connect clients? This doesn't make sense to me on face value, just checking

Comment (smaye81) [reply private/pkg/cert/certserver/certserver.go:36]: Talked offline to Mike.  It's not needed.  Closing this PR.
